staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1264,22 +1264,22 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 		// after adding the initial pods which is then used to establish a new watch request
 		useCurrentRV bool
 
+		// The test heavily relies on the fact that initialPods in created
```

Comment:
I think that after your change the tests should not rely on that, right ? Should we change the test cases so that we add some variation ?

Reply:
TBH - I completely can't remember why I added this...

Isn't it your PR that is doing that?

Reply:
I think you can remove the comment. Since the initial events are read from the map the result will always be random - unless we sort it which is exactly what this PR does. I updated/improved `RunWatchSemanticInitialEventsExtended`.

Reply:
done